google/cloud/spanner/integration_tests/client_integration_test.cc

```diff
@@ -404,6 +404,91 @@ TEST_F(ClientIntegrationTest, ExecuteSql) {
   EXPECT_THAT(actual_rows, UnorderedElementsAreArray(expected_rows));
 }
 
+void CheckReadWithOptions(
+    Client client,
+    std::function<Transaction::SingleUseOptions(CommitResult const&)> const&
+        options_generator) {
+  using RowType = Row<std::int64_t, std::string, std::string>;
+  std::vector<RowType> expected_rows;
+
+  auto commit = RunTransaction(
+      client, Transaction::ReadWriteOptions{},
+      // NOLINTNEXTLINE
+      [&expected_rows](Client, Transaction) -> StatusOr<Mutations> {
```

Comment:
Using `(Client const&, Transaction const&)` here should avoid the need for `NOLINT`.